api/core/model_runtime/model_providers/anthropic/llm/llm.py

```diff
@@ -526,12 +526,23 @@ def _convert_prompt_messages(self, prompt_messages: Sequence[PromptMessage]) ->
                                         f"Unsupported document type {message_content.mime_type}, "
                                         "only support application/pdf"
                                     )
+                                if not message_content.base64_data:
+                                    try:
+                                        document_content = requests.get(message_content.url).content
+                                        base64_data = base64.b64encode(document_content).decode("utf-8")
+                                    except Exception as ex:
+                                        raise ValueError(
+                                            f"Failed to fetch document data from url {message_content.url}, {ex}"
+                                        )
+                                else:
+                                    base64_data = message_content.base64_data
+
                                 sub_message_dict = {
                                     "type": "document",
                                     "source": {
                                         "type": "base64",
                                         "media_type": message_content.mime_type,
-                                        "data": message_content.data,
```

Comment:
You can use `message_content.base64_data` here.

Reply:
Thanks for the hint! I also try to do the same thing when handling images and no issues found. Please advise me if I should fix that as well.

Reply:
Claude always requires base64 content of the document, but the image supports base64 data and URL.